proj/pred/templates/pred/base_layout.html

```diff
@@ -70,7 +74,86 @@
 function ChangeImgSubmitNumSeq(imgPtr) {
     document.getElementById('imgNumSeqSubmit').src = ImgArraySubmitNumSeq[imgPtr];
 }
-
+const serverStatistics = JSON.parse("{{ statistics_per_timeline|escapejs }}")
+
+function drawGraph(id, type) {
+    const transformDate = (fullDate, id) => {
```

Comment:
I think there is an error here; `2022-01-01` and `2022-12-31` get the same title (`2022-W52`). I suppose the first one should belong to `2021-W52`.

Reply:
Good thing that you caught this! I think 03189a3 should fix it, have a look and see if I have missed anything

Reply:
I think the new code looks very nice, and it's nice that it's less complicated!
But as far as I can tell, the bug is still there, since it's the year that should be changed (eg  if the first week of the year is 52). Some potential solutions:
- add `if` statements, like you mentioned in the meeting
- use simplified (and slightly incorrect) week numbers, so that 1/1 always belongs to week 1 (`2022-01-01` -> `2022-W1`)

Reply:
I think the new code looks very nice, and it's nice that it's less complicated!
But as far as I can tell, the bug is still there, since it's the year that should be changed (eg  if the first week of the year is 52). Some potential solutions:
- add `if` statements, like you mentioned in the meeting
- use simplified (and slightly incorrect) week numbers, so that 1/1 always belongs to week 1 (`2022-01-01` -> `2022-W1`)

Reply:
44b2236371cc3bfc3f1819af0c5167b80dba60e1 This should do the trick!

Reply:
The current code still convert `2022-01-01` as  `2021-W52` as I tested. But I think it makes sense that `2022-01-01` is considered as `2021-W52` since the beginning of this week is from 2021. 

The refactored looks cleaner and easier to understand, well done!

```javascript
Date.prototype.getWeek = function() {s
    let d = new Date(Date.UTC(this.getFullYear(),
            this.getMonth(), this.getDate()));
    const dayNum = d.getUTCDay() || 7;
    d.setUTCDate(d.getUTCDate() + 4 - dayNum);
    const yearStart = new Date(Date.UTC(d.getUTCFullYear(),0,1));
    return `${yearStart.getFullYear()}-W${Math.ceil((((d - yearStart) / 86400000) + 1)/7)}`
}

console.log(new Date("2022-01-01").getWeek())
console.log(new Date("2022-01-03").getWeek())
console.log(new Date("2021-12-31").getWeek())
```
```
> "2021-W52"
> "2022-W1"
> "2021-W52"
```

Reply:
Thanks for the verification!